### PR TITLE
LPD-85711 Enable opt-in search functionality in Picker component

### PR DESCRIPTION
**Pull request by limaagabriel**
Ticket: https://liferay.atlassian.net/browse/LPD-85709

#### Summary
This PR introduces a real-time search filter within the Picker component's panel. This feature helps users efficiently locate options in large lists. The implementation is fully opt-in to ensure no impact on existing component consumers.

#### Changes
- **Logic:** Integrated case-insensitive partial matching into the useCollection hook used by Picker.
- **API:** Added searchable (boolean) and searchableThreshold (number) props to control the search UI.
- **UI:** Added a search input styled with Lexicon patterns at the top of the dropdown menu.
- **UX:** Implemented automatic focus on the search input upon opening the panel and maintained keyboard navigation support (ArrowKeys, Enter, Esc, PageUp/Down).
- **Tests:** Created a new test suite Search.tsx covering all search-related scenarios and edge cases.
- **Docs:** Updated picker.mdx documentation and added a Search story to the Storybook.

#### How to test it
1. Navigate to modules/apps/frontend-js/frontend-js-clay-web.
2. Run unit tests: yarn test clay/clay-core/src/picker/__tests__/Search.tsx. Verify all tests pass.
3. Run Storybook: yarn storybook.
4. Find the Design System > Components > Picker > Search story.
5. Verify that:
    - The 'Searchable (Explicit)' picker shows the search bar immediately.
    - The 'Automatic (Threshold 5)' picker shows the search bar since it has more than 5 items.
    - Typing in the search bar filters the list in real-time.
    - Keyboard navigation (arrows and Enter) works while the search input is focused.

image

### CODE REVIEW

File: modules/apps/frontend-js/frontend-js-clay-web/clay/clay-core/src/picker/Picker.tsx [hunk 1]
```diff
@@ -574,6 +644,86 @@ export function Picker<T extends Record<string, any> | string | number>({
...
+													}
+
+													if (
+														event.key === Keys.Esc
+													) {
+														setActive(false);
+														triggerRef.current?.focus();
+													}
+
+													if (
```
Comment (ambrinchaudhary) [inline modules/apps/frontend-js/frontend-js-clay-web/clay/clay-core/src/picker/Picker.tsx]: Should we also include `Keys.Up`/`Keys.Down` events?

**Review by ambrinchaudhary — APPROVED** [review]
LGTM! great job 

The only thing that "worries" me is to have inlined all the key events, I would prefer a separate function. But this is currently happening in the component, so this doesn't block this pull.

File: modules/apps/frontend-js/frontend-js-clay-web/clay/clay-core/src/picker/Picker.tsx (hunk 1 above)
Comment (limaagabriel) [reply modules/apps/frontend-js/frontend-js-clay-web/clay/clay-core/src/picker/Picker.tsx]: The arrow up and arrow down are already handled by the `useNavigation` hook

**Comment by limaagabriel** [comment]
Hi, @ambrinchaudhary! Regarding your comment about being inlined, I found out a way of keeping the search-related code more organized. The search logic is inside the `useSearch` hook, while the UI is in the `Search` component. Let me know what you think. Also found out some possible accessibility issues, so added a new commit addressing that as well.

File: modules/apps/frontend-js/frontend-js-clay-web/clay/clay-core/src/picker/Picker.tsx [hunk 2]
```diff
@@ -644,85 +634,20 @@ export function Picker<T extends Record<string, any> | string | number>({
...
-												tag="span"
-											>
-												<Icon symbol="search" />
-											</ClayInput.GroupInsetItem>
-										</ClayInput.GroupItem>
-									</ClayInput.Group>
-								</div>
-							</form>
-						)}
+						<Search
```
Comment (ambrinchaudhary) [inline modules/apps/frontend-js/frontend-js-clay-web/clay/clay-core/src/picker/Picker.tsx]: Remove `visible` prop and keep logic in this component: 

{isSearchable && (
    <Search .../>
)}

File: modules/apps/frontend-js/frontend-js-clay-web/clay/clay-core/src/picker/Search.tsx [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+	return (
+		visible && (
+			<div className="pb-2 pt-3 px-3">
+				<ClayInput.Group small>
+					<ClayInput.GroupItem className="input-group-item-focusable">
+						<ClayInput
+							aria-label={messages.searchPlaceholder}
+							insetAfter
+							onChange={onChange}
```
Comment (ambrinchaudhary) [inline modules/apps/frontend-js/frontend-js-clay-web/clay/clay-core/src/picker/Search.tsx]: To me it is cleaner if you inline

`onChange={(event) => onChange(event.target.value)}`

File: modules/apps/frontend-js/frontend-js-clay-web/clay/clay-core/src/picker/Picker.tsx [hunk 4]
```diff
@@ -644,85 +634,20 @@ export function Picker<T extends Record<string, any> | string | number>({
...
+							activeDescendant={activeDescendant}
+							getOptions={getOptions}
+							messages={messages}
```
Comment (ambrinchaudhary) [inline modules/apps/frontend-js/frontend-js-clay-web/clay/clay-core/src/picker/Picker.tsx]: We only need the `searchPlaceholder`, can we just pass it instead of full `messages` object?

**Review by ambrinchaudhary — CHANGES_REQUESTED** [review]
@limaagabriel I like more this approach. I suggested some changes, let me know what do you think.

**Comment by limaagabriel** [comment]
Thanks @ambrinchaudhary! Let me know if it looks better now

**Review by ambrinchaudhary — APPROVED** [review]

**Comment by ambrinchaudhary** [comment]
ci:forward

**Comment by limaagabriel** [comment]
ci:forward

**Comment by limaagabriel** [comment]
ci:test:sf

**Comment by limaagabriel** [comment]
ci:forward
